### ISSUE TITLE
[DOC] show where you have coded multivariate time series algorithms , if you have it ?

### ISSUE THREAD

**Issue by Sandy4321** [aeon-toolkit/aeon#589]
### Describe the issue linked to the documentation

I can not find multivariate time series classification, do you have one ?

![image](https://github.com/aeon-toolkit/aeon/assets/11426119/93695c63-3ede-449f-8f14-80e9b68d8dfc)
 

### Suggest a potential alternative/fix

show where you have coded multivariate time series algorithms , if you have it ?

**Comment by TonyBagnall** [comment]
hi, depends on what task you are looking at. From here
https://www.aeon-toolkit.org/en/latest/examples/classification/classification.html

A list of classifiers capable of handling multivariate classification can be obtained with this code
```python
from aeon.registry import all_estimators

all_estimators(
    filter_tags={"capability:multivariate": True}, estimator_types="classifier"
)
```
similar model for clustering and regression.

**Comment by TonyBagnall** [comment]
hi @Sandy4321 did you work it out? Happy to provide further advice, but if its good could you close this PR? thanks

**Comment by Sandy4321** [comment]
thanks for willing to help
it is what I get 
from aeon.registry import all_estimators
all_estimators(
    filter_tags={"capability:multivariate": True}, estimator_types="classifier"
)

Traceback (most recent call last):
  Debug Console, prompt 42, line 1
builtins.ModuleNotFoundError: No module named 'aeon'
from aeon.registry import all_estimators

all_estimators(
    filter_tags={"capability:multivariate": True}, estimator_types="classifier"
)

C:\my_py_environments\py310_torch_cuda_july2023\lib\site-packages\aeon\utils\validation\_dependencies.py:142: UserWarning: No module named 'pycatch22'. 'pycatch22' is a soft dependency and not included in the base aeon installation. Please run: `pip install pycatch22` to install the pycatch22 package. To install all soft dependencies, run: `pip install aeon[all_extras]`
  warnings.warn(msg)
C:\my_py_environments\py310_torch_cuda_july2023\lib\site-packages\numba\core\decorators.py:282: RuntimeWarning: nopython is set for njit and is ignored
  warnings.warn('nopython is set for njit and is ignored', RuntimeWarning)
C:\my_py_environments\py310_torch_cuda_july2023\lib\site-packages\aeon\utils\validation\_dependencies.py:142: UserWarning: No module named 'esig'. 'esig' is a soft dependency and not included in the base aeon installation. Please run: `pip install esig` to install the esig package. To install all soft dependencies, run: `pip install aeon[all_extras]`
  warnings.warn(msg)
C:\my_py_environments\py310_torch_cuda_july2023\lib\site-packages\aeon\utils\validation\_dependencies.py:142: UserWarning: No module named 'statsforecast'. 'statsforecast' is a soft dependency and not included in the base aeon installation. Please run: `pip install statsforecast` to install the statsforecast package. To install all soft dependencies, run: `pip install aeon[all_extras]`
  warnings.warn(msg)


my OS is windows

and I had some errors during installation

Successfully built filterpy databricks-cli pymeeus
Installing collected packages: pymeeus, korean-lunar-calendar, ephem, dm-tree, waitress, tensorflow-probability, sqlparse, querystring-parser, pyarrow, keras-self-attention, importlib-resources, hijri-converter, entrypoints, deprecated, convertdate, blinker, stumpy, scikit-learn, pooch, LunarCalendar, holidays, Flask, docker, databricks-cli, tslearn, pyod, mne, mlflow, kotsu, hmmlearn, gluonts, filterpy, cmdstanpy, aeon, tsfresh, scikit-posthocs, prophet, pmdarima, tbats, tensorflow
  Attempting uninstall: pyarrow
    Found existing installation: pyarrow 12.0.1
    Uninstalling pyarrow-12.0.1:
      Successfully uninstalled pyarrow-12.0.1
  Attempting uninstall: scikit-learn
    Found existing installation: scikit-learn 1.3.0
    Uninstalling scikit-learn-1.3.0:
      Successfully uninstalled scikit-learn-1.3.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pytorch-forecasting 1.0.0 requires torch<3.0.0,>=2.0.0, but you have torch 1.13.1 which is incompatible.
Successfully installed Flask-2.3.2 LunarCalendar-0.0.9 aeon-0.3.0 blinker-1.6.2 cmdstanpy-1.1.0 convertdate-2.4.0 databricks-cli-0.17.7 deprecated-1.2.14 dm-tree-0.1.8 docker-6.1.3 entrypoints-0.4 ephem-4.1.4 filterpy-1.4.5 gluonts-0.13.2 hijri-converter-2.3.1 hmmlearn-0.3.0 holidays-0.24 importlib-resources-6.0.0 keras-self-attention-0.51.0 korean-lunar-calendar-0.3.1 kotsu-0.3.2 mlflow-2.3.2 mne-1.4.2 pmdarima-2.0.3 pooch-1.7.0 prophet-1.1.3 pyarrow-11.0.0 pymeeus-0.5.12 pyod-1.1.0 querystring-parser-1.2.4 scikit-learn-1.2.2 scikit-posthocs-0.7.0 sqlparse-0.4.4 stumpy-1.11.1 tbats-1.1.3 tensorflow-2.13.0 tensorflow-probability-0.20.1 tsfresh-0.20.1 tslearn-0.6.1 waitress-2.1.2

---------------------------------------------------------
to sum up , may you at least for now share what result should be for 
from aeon.registry import all_estimators
all_estimators(
    filter_tags={"capability:multivariate": True}, estimator_types="classifier"
)
or just list of 
classifiers, those support  multivariate time series ?

**Comment by Sandy4321** [comment]
I tried to install 
pip install aeon[all_extras]

**Comment by Sandy4321** [comment]
for 
pip install statsforecast
I get 
Installing collected packages: trace-updater, sqlglot, dash-table, dash-html-components, dash-core-components, antlr4-python3-runtime, Werkzeug, tsdownsample, retrying, orjson, fs, ansi2html, Flask, triad, dash, plotly-resampler, fugue-sql-antlr, adagio, qpd, fugue, statsforecast
  Attempting uninstall: antlr4-python3-runtime
    Found existing installation: antlr4-python3-runtime 4.9.3
    Uninstalling antlr4-python3-runtime-4.9.3:
      Successfully uninstalled antlr4-python3-runtime-4.9.3
  Attempting uninstall: Werkzeug
    Found existing installation: Werkzeug 2.3.6
    Uninstalling Werkzeug-2.3.6:
      Successfully uninstalled Werkzeug-2.3.6
  Attempting uninstall: Flask
    Found existing installation: Flask 2.3.2
    Uninstalling Flask-2.3.2:
      Successfully uninstalled Flask-2.3.2
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
omegaconf 2.3.0 requires antlr4-python3-runtime==4.9.*, but you have antlr4-python3-runtime 4.11.1 which is incompatible.
Successfully installed Flask-2.2.5 Werkzeug-2.2.3 adagio-0.2.4 ansi2html-1.8.0 antlr4-python3-runtime-4.11.1 dash-2.11.1 dash-core-components-2.0.0 dash-html-components-2.0.0 dash-table-5.0.0 fs-2.4.16 fugue-0.8.6 fugue-sql-antlr-0.1.6 orjson-3.9.2 plotly-resampler-0.9.1 qpd-0.4.4 retrying-1.3.4 sqlglot-17.8.0 statsforecast-1.5.0 trace-updater-0.0.9.1 triad-0.9.1 tsdownsample-0.1.2

**Comment by Sandy4321** [comment]
1
should I create new clear python environment and try to run?
pip install aeon[all_extras]
2
may you share  just list of classifiers, those support multivariate time series ?

**Comment by TonyBagnall** [comment]
For estimator_types="regressor" and "clusterer" you get

```python-trace
 Multivariate regressors
('CNNRegressor', <class 'aeon.regression.deep_learning.cnn.CNNRegressor'>)
('CanonicalIntervalForestRegressor', <class 'aeon.regression.interval_based._cif.CanonicalIntervalForestRegressor'>)
('Catch22Regressor', <class 'aeon.regression.feature_based._catch22.Catch22Regressor'>)
('DrCIFRegressor', <class 'aeon.regression.interval_based._drcif.DrCIFRegressor'>)
('DummyRegressor', <class 'aeon.regression._dummy.DummyRegressor'>)
('FCNRegressor', <class 'aeon.regression.deep_learning.fcn.FCNRegressor'>)
('FreshPRINCERegressor', <class 'aeon.regression.feature_based._fresh_prince.FreshPRINCERegressor'>)
('InceptionTimeRegressor', <class 'aeon.regression.deep_learning.inception_time.InceptionTimeRegressor'>)
('IndividualInceptionRegressor', <class 'aeon.regression.deep_learning.inception_time.IndividualInceptionRegressor'>)
('IntervalForestRegressor', <class 'aeon.regression.interval_based._interval_forest.IntervalForestRegressor'>)
('KNeighborsTimeSeriesRegressor', <class 'aeon.regression.distance_based._time_series_neighbors.KNeighborsTimeSeriesRegressor'>)
('RandomIntervalRegressor', <class 'aeon.regression.interval_based._interval_pipelines.RandomIntervalRegressor'>)
('RandomIntervalSpectralEnsembleRegressor', <class 'aeon.regression.interval_based._rise.RandomIntervalSpectralEnsembleRegressor'>)
('ResNetRegressor', <class 'aeon.regression.deep_learning.resnet.ResNetRegressor'>)
('RocketRegressor', <class 'aeon.regression.convolution_based._rocket_regressor.RocketRegressor'>)
('TapNetRegressor', <class 'aeon.regression.deep_learning.tapnet.TapNetRegressor'>)
('TimeSeriesForestRegressor', <class 'aeon.regression.interval_based._tsf.TimeSeriesForestRegressor'>)
 Multivariate clusterers
('TimeSeriesCLARA', <class 'aeon.clustering.clara.TimeSeriesCLARA'>)
('TimeSeriesCLARANS', <class 'aeon.clustering.clarans.TimeSeriesCLARANS'>)
('TimeSeriesKMeans', <class 'aeon.clustering.k_means.TimeSeriesKMeans'>)
('TimeSeriesKMedoids', <class 'aeon.clustering.k_medoids.TimeSeriesKMedoids'>)
('TimeSeriesKShapes', <class 'aeon.clustering.k_shapes.TimeSeriesKShapes'>)
('TimeSeriesKernelKMeans', <class 'aeon.clustering.kernel_k_means.TimeSeriesKernelKMeans'>)
```

**Comment by TonyBagnall** [comment]
not sure about the environment, sorry, maybe someone else can help with that (but a clean environment sounds sensible)

**Comment by Sandy4321** [comment]
great then 
from aeon.registry import all_estimators
names = all_estimators(
filter_tags={"capability:multivariate": True}, estimator_types="classifier"
)
print(" Multivariate classifiers")
for n in names:
    print(n)

which one can handle categorical values for features

**Comment by TonyBagnall** [comment]
we don't handle categorical time series, sorry, whole different thing.

**Comment by Sandy4321** [comment]
I see, do you know other python packages can do it : handle multivariate categorical time series

**Comment by TonyBagnall** [comment]
none I am aware of, but Im sure there is something out there somewhere

**Comment by MatthewMiddlehurst** [comment]
Closing.
